src/main/java/com/spanish_inquisition/battleship/server/database/DatabaseController.java

```diff
@@ -0,0 +1,100 @@
+package com.spanish_inquisition.battleship.server.database;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by michal on 08.08.17.
+ */
+public class DatabaseController {
+    static String JDBC_DRIVER = "org.sqlite.JDBC";
+    static final String DB_NAME = "history.db";
+    static final String TABLE_NAME = "players";
+    static String DATABASE = "jdbc:sqlite:dbs/" + DB_NAME;
+
+    void saveOrUpdatePlayerEntity(PlayerEntity playerEntity) {
+        if(checkIfPlayerExists(playerEntity, loadPlayers())) {
+            updatePlayer(playerEntity);
+        } else {
+            savePlayer(playerEntity);
+        }
+    }
+
+    private static void loadJDBCDriver() {
+        try {
+            Class.forName(JDBC_DRIVER);
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void savePlayer(PlayerEntity playerEntity) {
+        loadJDBCDriver();
+        try (Connection connection = DriverManager.getConnection(DATABASE);
+             Statement statement = connection.createStatement()) {
+            createTableIfNotExists(statement);
+            PreparedStatement preparedStatement = connection.prepareStatement(String.format("insert into %s (name, score) values (?, ?);", TABLE_NAME));
+            setSaveStatementStrings(preparedStatement, playerEntity);
+            preparedStatement.executeBatch();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void updatePlayer(PlayerEntity playerEntity) {
+        loadJDBCDriver();
+        try (Connection connection = DriverManager.getConnection(DATABASE);
+             Statement statement = connection.createStatement()) {
+            createTableIfNotExists(statement);
```

Comment:
You don't have to execute this here - if updatePlayer is called - you are sure that table exists.